### ISSUE TITLE
Questions about NCCL_NVB_DISABLE

### ISSUE THREAD

**Issue by MonroeD** [NVIDIA/nccl#819]
I see in the nccl doc NCCL_NVB_DISABLE means "Disable intra-node communication through NVLink via an intermediate GPU".
is that means as followed.
1. gpu0 cannot connect direct with gpu5 using nvlink, and gpu0 can connect gpu5 through gpu1 as a hop using nvlink;
2. so if gpu0 send data to gpu5, can the data send to gpu5 through gpu1? that means nccl send data to gpu1 first, and then gpu1 send data to gpu5

**Comment by sjeaugey** [comment]
Yes that's the idea. Gpu1 will not really "send" data to gpu5; gpu5 will read from gpu1 instead. The only difference with NVB operation compared to normal operation is that the memory used for communication is not on the receiver GPU (and sender GPU for credit return) but on the intermediate GPU, which both sender and receiver GPUs can access directly. The intermediate GPU is passive, aside from allocating freeing the intermediate buffer.

**Comment by MonroeD** [comment]
Thank. the explanation is so good! I have another question. Is NCCL_NVB_DISABLE only affect ncclSend/ncclRecv operation?

**Comment by MonroeD** [comment]
Got it. if we can build a ring using NVB then we can use any other algorithms(reduce, all_reduce, broadcast. all_gather, scatter)

**Comment by stephenmsachs** [comment]
### Issue Cleanup: Helping Us Focus on Current Challenges

  We're reviewing older issues to ensure we prioritize the most relevant and active ones. Since this issue hasn't seen updates in over 6 months, we'll be closing it for now.

  ,*This change helps us focus our efforts on addressing any current issues our users are facing.* If this issue still affects you, please don't hesitate to reopen it with a quick update (e.g., "Still relevant on [version=X]").
  Thanks for your understanding and for contributing to NCCL.
